### PR TITLE
10 Improve template variants formatting

### LINKED ISSUE #10: Improve template variants formatting
Problem: Currently we have only the main template abbreviations in the left hand side index, e.g. 
image
However, for some we have variants, e.g. `CMT2A`. This currently shows like this:
image
inside the `CMT2` page. 

To do: Improve the formatting of these template variants.
Ideas:
- [x] Can we have a subitem in the index? Such that `CMT2A` would appear below `CMT2` and clicking on it would open a separate page?
   - [x] try out first for one template
   - [x] we could then factor out `CMT2A` as a separate qmd file
   - [x] to avoid duplicate setup code, we could use the `include` as in https://github.com/insightsengineering/biomarker-catalog/blob/main/graphs/dg02.qmd#LL88C1-L88C1
   - [x] advantage is that we could also directly link to a subvariant
   - [x] Check how to subitemize variants in the sidebar: if done by creating subfolders and indenting, update the relative URLs
   - [x] correct relative paths to the main templates (without A/B/C..) in the index as well (bac-index.qmd)
   - [x] if we are fine with this variant, based on CMT2 example, then implement the same for the other around 37 templates with A/B/C/etc variants
   - [x] Rename "Standard Plot" to "Plot" where only one plot tab remains
 - [x] Index: Change to hierarchical structure `DG`, then underneath `DG1-4`
 - [x] Navigation: Have 3 levels `DG`, then underneath `DG1-4`, then underneath the variants
 - [x] When all templates implemented in the new hierarchy, check again all cross-links (relative URLs) to make sure nothing is broken (eg RFG3 updated before SFG4)
 

~~- [ ] Can we just shorten the tab titles? E.g. in the `CMT2A` example we could have "Changing Reference Levels" even without the `CMT2A`, which is anyway not really used or standard (this is how the biomarker catalog is doing that)~~
   ~~- [ ] note that the downside of the current structure is that we cannot link directly to these variants. We can write something like `https://insightsengineering.github.io/biomarker-catalog/graphs/rfg01.html#L60` as the URL but page still opens on the Setup tab~~
   ~~- [ ] as a workaround, seems there could be a JavaScript solution, see https://stackoverflow.com/questions/66246651/bookdown-cross-referencing-content-in-tabbed-section (@insightsengineering/idr fyi)~~

### PR DESCRIPTION
**Pull request by moronip81**
closes #10

### CODE REVIEW

**Comment by moronip81** [comment]
Hi @danielinteractive this is meant to be the final PR

File: bac_index.qmd [hunk 1]
```diff
@@ -32,3 +32,32 @@ toc-depth: 4
     -   [KG3 -- Kaplan-Meier Graphs by Biomarker Subgroups](./graphs/kg03.qmd)
     -   [KG4 -- Kaplan-Meier Graphs by Treatment Arm and Biomarker Subgroups](./graphs/KG4/kg04.qmd)
     -   [KG5 -- Kaplan-Meier Graphs by Treatment Arm and Continuous Biomarker Cut](./graphs/KG5/kg05.qmd)
+-   [RFG -- Response Forest Graphs](./graphs/RFG1/rfg01.qmd)
```
Comment (danielinteractive) [inline bac_index.qmd:35]: how about have sections in this index "Tables" and "Graphs" ? since we also have that split in the navigation pane

**Review by danielinteractive — COMMENTED** [review]
Thanks @moronip81 ! looks nice. On the review tab just seeing `dg04.qmd` looking a bit out of place potentially, maybe double check its location?

File: bac_index.qmd (hunk 1 above)
Comment (moronip81) [reply bac_index.qmd:35]: Hi @danielinteractive: I did not move from the tables/graphs main folder all the templates that did not have A/B/etc variants, this is the case of dg04.qmd but also others, this aims at keep the structure and relative URLs simpler, if needed can be changed (requires revising all cross-links)
About the index, I actually see the same split like in the nav pane (maybe have a look to the rendered website), or do you mean anything else?
```
### Tables

-   [CMT -- Cox Proportional Hazards Model Tables](./tables/cmt01.qmd)
    -   [CMT1 -- Cox Proportional Hazards Model Tables with a Single Covariate](./tables/cmt01.qmd)
    -   [CMT2 -- Cox Proportional Hazards Model Tables with Multiple Covariates](./tables/CMT2/cmt02.qmd)
    -   [CMT3 -- Cox Proportional Hazards Model Tables with Separate Models for Each Covariate](./tables/cmt03.qmd)
-   [DT -- Demographics Tables](./tables/DT1/dt01.qmd)
    -   [DT1 -- Demographics Tables Comparing BEP vs. Overall Population](./tables/DT1/dt01.qmd)
    -   [DT2 -- Demographics Tables Comparing BEP vs. non-BEP Population](./tables/DT2/dt02.qmd)
-   [TET -- Time-to-Event Summary Tables](./tables/TET1/tet01.qmd)
    -   [TET1 -- Time-to-Event Summary Tables](./tables/TET1/tet01.qmd)

### Graphs

-   [AG -- Utilities for All Graphs](./graphs/ag01.qmd)```

Comment (danielinteractive) [reply bac_index.qmd:35]: oh sorry, I overlooked the Tables and Graphs section titles... so all good!

**Review by danielinteractive — APPROVED** [review]

**Pull request merged**